app/api/v1/endpoints/control/attacks.py

```
@@ -0,0 +1,769 @@
+    current_user: Annotated[User, Depends(get_current_control_user)],
+) -> AttackOut:
+    """
+    Create a new attack.
+
+    The user must have access to the campaign's project.
+    Referenced resources (wordlist, rules, masks) must exist.
+    """
+    try:
+        # Validate campaign access
+        if data.campaign_id is None:
+            raise CampaignNotFoundProblem(detail="campaign_id is required")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)